src/client/clientlauncher.cpp

```diff
@@ -498,6 +515,12 @@ bool ClientLauncher::launch_game(std::string &error_message,
 
 void ClientLauncher::main_menu(MainMenuData *menudata)
 {
+#if USE_DISCORD
+	g_pDiscord->setDetails("In Main Menu");
+	g_pDiscord->setState("");
+	g_pDiscord->updatePresence();
```

Comment:
for me g_pDiscord->updatePresence(); should be called in setDetails

Reply:
No agree. Subsequently we can change a lot of things (look at my DataRichPresence structure (and still i have not marked everything in it))

Discord_UpdatePresence() has a rate limit of one update per 15 seconds. 
So it's better to call the update when we really finished updating all our data. ;)

Reply:
if there is a rate limit discord object should handle that by having a flag to know if it should update or not to handle the error case

Reply:
The SDK will queue up any presence updates sent in that window and send the newest one once the client is free to do so.
So you prefer that we update each time ?